apps/experiments/views/prompt.py

```diff
@@ -4,17 +4,107 @@
 
 from celery.result import AsyncResult
 from celery_progress.backend import Progress
-from django.http import JsonResponse
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import get_object_or_404
 from django.template.response import TemplateResponse
 from django.urls import reverse
 from django.utils import timezone
 from django.views.decorators.http import require_POST
+from django.views.generic import CreateView, UpdateView
+from django_tables2 import SingleTableView
 
 from apps.experiments.helpers import get_real_user_or_none
 from apps.experiments.models import Prompt, PromptBuilderHistory, SourceMaterial
+from apps.experiments.tables import PromptTable
 from apps.experiments.tasks import get_prompt_builder_response_task
 from apps.teams.decorators import login_and_team_required, team_admin_required
 
+PROMPT_DATA_SESSION_KEY = "prompt_data"
+
+
+@login_and_team_required
+def prompt_home(request, team_slug: str):
+    return TemplateResponse(
+        request,
+        "generic/object_home.html",
+        {
+            "active_tab": "prompts",
+            "title": "Prompts",
+            "new_object_url": reverse("experiments:prompt_new", args=[team_slug]),
+            "table_url": reverse("experiments:prompt_table", args=[team_slug]),
+        },
+    )
+
+
+class PromptTableView(SingleTableView):
+    model = Prompt
+    paginate_by = 25
```

Comment:
Out of curiosity, does this handle pagination for us?

Reply:
It's handled by [django-tables2](https://django-tables2.readthedocs.io/en/latest/pages/pagination.html). You can see what it's doing in `django_tables2.views.TableMixinBase.get_table_pagination`.